clustering/web/undertow/src/main/java/org/wildfly/clustering/web/undertow/session/DistributableSessionManager.java

```diff
@@ -76,17 +85,61 @@ public SessionListeners getSessionListeners() {
 
     @Override
     public synchronized void start() {
+        this.lifecycleStamp.ifPresent(this);
         this.manager.start();
         if (this.statistics != null) {
             this.statistics.reset();
         }
     }
 
+    @Override
+    public void accept(long stamp) {
+        this.lifecycleLock.unlock(stamp);
+        this.lifecycleStamp = OptionalLong.empty();
```

Comment:
Should this be in a synchronized block?

Reply:
This method is already called from within a synchronized block.  The only reason this is a separate public method is to avoid the cost of a lambda or anonymous class.